### ISSUE TITLE
How to improve `imresize` for downsizing

### ISSUE THREAD

**Issue by Evizero** [JuliaImages/ImageTransformations.jl#28]
Right now `imresize` performs something that is pretty similar to simple pixel sub-sampling. Naturally the quality of the result isn't particularly breathtaking.

I open this issue to brainstorm on what would be a good way to improve the quality for downsizing and image to an arbitrary specified size.

The straight forward way of doing it would be in two steps: 1.) Filter image, 2.) sub-sample. However, this would add an additional temporary array that I suspect could probably be avoided.

**Comment by timholy** [comment]
[This page](http://entropymine.com/imageworsener/resample/) points out that one can effectively use the same splines that we use for interpolation, but stretch them to cover the "grid" of the reduced-size image. Not entirely trivial to implement, but not insanely difficult either. I suspect the main question is whether we want to support multiple splines or just pick one. I suspect if we started with linear (triangular splines) we'd be pretty happy.

Interestingly, that page suggests handling one dimension at a time, rather than computing the result directly. It's the same strategy that `restrict` is using.

**Comment by Evizero** [comment]
Thanks for the pointers. I think this is currently a little beyond me in terms of skill and time, but a good thing to keep in mind.

**Comment by babaq** [comment]
I am also curious about interpolations for upsampling/downsampling of images, the MATLAB and Pillow both support several methods, and some of them, for example , the Lanczos kernel seems easy to implement in the image filtering framework, is it possible to just add the kernel, and then, we can use filtering and imresize to achieve anti-aliasing resizing?

**Comment by johnnychen94** [comment]
> I am also curious about interpolations for upsampling/downsampling of images, the MATLAB and Pillow both support several methods, and some of them, for example , the Lanczos kernel seems easy to implement in the image filtering framework, is it possible to just add the kernel, and then, we can use filtering and imresize to achieve anti-aliasing resizing?

This is now available (provided by #115) since v0.8.11:

```julia
using Interpolations

imresize(img, sz; method=Lanczos(2))
imresize(img, sz; method=Lanczos4OpenCV())
```

**Comment by johnnychen94** [comment]
I'm closing this in favor of #120 as I didn't observe any particular bad alias result.

But feel free to reopen it if this is still an issue.
